src/provider/view/components/NotificationView/NotificationView.tsx

```diff
@@ -92,15 +95,19 @@ function getGroupTitle(notification: StoredNotification, groupingType: GroupingT
 /**
  * Group notifications together based on the given grouping method e.g. application/date.
  * @param notifications List of notifications to sort into groups.
+ * @param applications A map of application registry to resolve the notification title
  * @param groupMethod Grouping type to use for sorting.
  */
-function groupNotifications(notifications: StoredNotification[], groupMethod: GroupingType): Map<string, Group> {
+function groupNotifications(notifications: StoredNotification[], applications: ApplicationMap, groupMethod: GroupingType): Map<string, Group> {
     // TODO: This sorting should be removed and changed to use insert-sort inside the reduce below
-    // Pre-sort notifications by date (groups will then also be sorted by date)
-    notifications.sort((a: StoredNotification, b: StoredNotification) =>
-        b.notification.date.valueOf() - a.notification.date.valueOf());
-    // Reduce the notifications array into a Map of notifications grouped by their title
-    const groupMap = notifications.reduce((groups: Map<string, Group>, currentNotification: StoredNotification) => {
+    const groupMap = notifications.map((notification) => {
+        // Map stored notifications to titled notifications
+        return {...notification, title: (applications.get(notification.source.uuid) || {title: notification.source.name || ''}).title};
+    }).sort((a: StoredNotification, b: StoredNotification) => {
+        // Sort notifications by date (groups will then also be sorted by date)
+        return b.notification.date.valueOf() - a.notification.date.valueOf();
+    }).reduce((groups: Map<string, Group>, currentNotification: TitledNotification) => {
+        // Reduce the titled notifications array into a Map of notifications grouped by their title
         const groupTitle = getGroupTitle(currentNotification, groupMethod);
```

Comment:
And below this - the line

    let key: string = currentNotification.source.name || currentNotification.source.uuid;

Should this be:

    let key: string = currentNotification.source.uuid;

? Do we want notifications from the same app but different windows to have different keys?

Reply:
I tried to plug in the titles without changing any existing logic within the UI as it didn't feel like within scope of this story. But makes sense, changed.